dae/dae/genomic_resources/gene_models.py

```diff
@@ -491,7 +492,11 @@ def gene_names(self):
         return list(self.gene_models.keys())
 
     def gene_models_by_gene_name(self, name):
-        return self.gene_models.get(name, None)
+        pattern = re.compile(re.escape(name) + "$", re.IGNORECASE)
+        for key in self.gene_models:
+            if pattern.match(key):
+                return self.gene_models[key]
+        return None
```

Comment:
I'm not really sure about this code... it turns a O(1) function into an O(n) one.